File: services/consultationservice/internal/model/post.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Post struct {
	ID       primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Title    string             `bson:"title" json:"title"`
	Content  string             `bson:"content" json:"content"`
	AuthorID string             `bson:"author_id" json:"author_id"`

	
	Tags       []string          `bson:"tags" json:"tags"`
	Categories []string          `bson:"categories" json:"categories"`
	Media      []MediaAttachment `bson:"media" json:"media"`
	Mentions   []string          `bson:"mentions" json:"mentions"`     
	Hashtags   []string          `bson:"hashtags" json:"hashtags"`     

	
	Visibility string `bson:"visibility" json:"visibility"` 
	PostType   string `bson:"post_type" json:"post_type"`   

	
	ViewCount     int `bson:"view_count" json:"view_count"`
	UpvoteCount   int `bson:"upvote_count" json:"upvote_count"`
	DownvoteCount int `bson:"downvote_count" json:"downvote_count"`
	CommentCount  int `bson:"comment_count" json:"comment_count"`
	ShareCount    int `bson:"share_count" json:"share_count"`

	
	UserVote     *string `bson:"-" json:"user_vote,omitempty"`     
	UserBookmark bool    `bson:"-" json:"user_bookmark,omitempty"` 

	
	IsDeleted bool `bson:"is_deleted" json:"is_deleted"`

	CreatedAt time.Time `bson:"created_at" json:"created_at"`
	UpdatedAt time.Time `bson:"updated_at" json:"updated_at"`
}

type MediaAttachment struct {
	Type    string `bson:"type" json:"type"`                         
	URL     string `bson:"url" json:"url"`
	Caption string `bson:"caption,omitempty" json:"caption,omitempty"`
}

File: services/mobileflutter/mobileapp/lib/models/post.dart
class Post {
  final String id;
  final String title;
  final String content;
  final String authorId;
  final String? authorName;
  final String? authorAvatar;
  final List<String> tags;
  final List<String> categories;
  final int upvoteCount;
  final int downvoteCount;
  final int commentCount;
  final String createdAt;
  final String updatedAt;

  Post({
    required this.id,
    required this.title,
    required this.content,
    required this.authorId,
    this.authorName,
    this.authorAvatar,
    required this.tags,
    required this.categories,
    required this.upvoteCount,
    required this.downvoteCount,
    required this.commentCount,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'] ?? '',
      title: json['title'] ?? '',
      content: json['content'] ?? '',
      authorId: json['author_id'] ?? '',
      authorName: json['author_name'],
      authorAvatar: json['author_avatar'],
      tags: List<String>.from(json['tags'] ?? []),
      categories: List<String>.from(json['categories'] ?? []),
      upvoteCount: json['upvote_count'] ?? 0,
      downvoteCount: json['downvote_count'] ?? 0,
      commentCount: json['comment_count'] ?? 0,
      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'content': content,
      'author_id': authorId,
      'author_name': authorName,
      'author_avatar': authorAvatar,
      'tags': tags,
      'categories': categories,
      'upvote_count': upvoteCount,
      'downvote_count': downvoteCount,
      'comment_count': commentCount,
      'created_at': createdAt,
      'updated_at': updatedAt,
    };
  }
}
